Plugins/DeleteUnusedBranches/DeleteUnusedBranchesForm.cs

```diff
@@ -212,15 +215,29 @@ private void Refresh_Click(object sender, EventArgs e)
             ThreadHelper.JoinableTaskFactory.RunAsync(() => RefreshObsoleteBranchesAsync());
         }
 
+        private void CheckBoxHeader_OnCheckBoxClicked(bool state)
+        {
+            for (int i = 0; i < BranchesGrid.Rows.Count; i++)
+            {
+                DataGridViewRow row = BranchesGrid.Rows[i];
+                DataGridViewCheckBoxCell cell =
+                    (DataGridViewCheckBoxCell)row.Cells[nameof(deleteDataGridViewCheckBoxColumn)];
+                cell.Value = state;
```

Comment:
follow the .NET event handlers design guidelines, and the column would be accessible via the `sender` argument